### PR TITLE
Added panic fix

### PR DESCRIPTION
**Pull request by jessicachango**

### CODE REVIEW

File: ujson.go [hunk 1]
```diff
@@ -6,17 +6,21 @@ import (
 )
 
 type JSON struct {
-	root interface{}
+	Root interface{}
```
Comment (mreiferson) [inline ujson.go]: what's the motivation for exposing this?  I would like to try to avoid leaking internals, should we just add an accessor method?

**Comment by jessicachango** [comment]
Sorry about the late response, I was away for 3 weeks.

The root is exposed because there were problems with variable type when trying to retrieving the value of an element from a dictionary of dictionaries, dictionary of arrays, array of dictionary and etc.

I agree that it's better if the root is not exposed and there should be a getter function instead. However, we already made quite a bit of changes to our code base assuming that the Root is exposed. I am going to keep the root not exposed in this pull request so your branch is good.

**Comment by mreiferson** [comment]
Thanks, np!

> I agree that it's better if the root is not exposed and there should be a getter function instead. However, we already made quite a bit of changes to our code base assuming that the Root is exposed. I am going to keep the root not exposed in this pull request so your branch is good.

Hmmm, I'd be bummed for us to diverge when we both agree on the details?  Should we just decide on a getter?  `Interface()`?

**Comment by jessicachango** [comment]
I like the idea of Interface()

``` golang
func (j *JSON) Interface() interface{} {
    return j.root
}
```

We run into problems with variable type when an element within an array inside a ujson decoded dictionary is interface. I want to add another function

``` golang
func (j *JSON) JSON() *JSON{
    return &JSON{ j.root}
}
```

**Comment by mreiferson** [comment]
> We run into problems with variable type when an element within an array inside a ujson decoded dictionary is interface. I want to add another function

Perhaps none of `go-ujson`'s methods should return anything other than `*JSON`?

**Comment by jessicachango** [comment]
as you wish :) I updated the pull request accordingly.

**Comment by mreiferson** [comment]
Looks great!  Do you mind squashing down to one commit?

Thanks!

**Comment by jessicachango** [comment]
Sorry about that. The commit has been squashed to 1. In addition, I added more check for index not exceeding the length of j.data.

**Comment by mreiferson** [comment]
:+1: thanks!

**Pull request merged**